src/tests/ftest/control/dmg_system_reformat.py

```diff
@@ -62,28 +62,40 @@ def test_dmg_system_reformat(self):
 
         # To verify that we are using the membership information instead of the
         # dmg config explicit hostlist
-        # Uncomment below after DAOS-5979 is resolved
-        # self.assertTrue(
-        #     self.server_managers[-1].dmg.set_config_value("hostlist", None))
+        self.assertTrue(
+            self.server_managers[-1].dmg.set_config_value("hostlist", None))
 
         self.log.info("Perform dmg storage format on all system ranks:")
-        self.get_dmg_command().storage_format(force=True)
-        if self.get_dmg_command().result.exit_status != 0:
-            self.fail("Issues performing storage format --force: {}".format(
-                self.get_dmg_command().result.stderr_text))
+
+        # Calling storage format after system stop too soon would fail, so
+        # wait 10 sec and retry up to 4 times.
+        count = 0
+        while count < 4:
+            try:
+                self.get_dmg_command().storage_format(force=True)
+                if self.get_dmg_command().result.exit_status != 0:
+                    self.fail(
+                        "Issues performing storage format --force: {}".format(
+                            self.get_dmg_command().result.stderr_text))
+                break
+            except CommandFailure as error:
+                self.log.info(
+                    "Storage format failed. Wait 10 sec and retry. %s", error)
+                count += 1
+                time.sleep(10)
 
         # Check that engine starts up again
         self.log.info("<SERVER> Waiting for the engines to start")
-        self.server_managers[-1].detect_engine_start(host_qty=2)
+        self.server_managers[-1].detect_engine_start()
```

Comment:
It seems that if all 4 tries fail, we would still call detect_engine_start(). But I guess the engines would fail to start if format failed.